### PR TITLE
Fix some `TFWhisperModelIntegrationTests`

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

(Probably since the introduction of `GenerationConfig`), some TF Whisper Integration tests fail with error 

```bash
ValueError: The following `model_kwargs` are not used by the model: ['language', 'task'] (note: typos in the generate arguments will also show up in this list)
```

From my understanding, we should pass some arguments via `generation_kwargs`.

Note that `WhisperForConditionalGeneartion` has its custom `generate` but `TFWhisperForConditionalGeneartion` doesn't.

⚠️ When I try to apply the same changes to PyTorch Whisper test methods, it fails as the output is different.

We are somehow in a trouble of the inconsistency between PT/TF.

(Not sure if we should overwrite `generate` in `TFWhisperForConditionalGeneartion`.)

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
LGTM, thanks for fixing!

**Review by amyeroberts — APPROVED** [review]
LGTM- thanks for fixing! 

Regarding the PyTorch tests failing with the same inputs - how do they fail?

**Comment by ydshieh** [comment]
> LGTM- thanks for fixing!
> 
> Regarding the PyTorch tests failing with the same inputs - how do they fail?

It gives different outputs

**Comment by ydshieh** [comment]
@gante I am going to merge now. But I know you will have some insights regarding this issue.

**Comment by ydshieh** [comment]
I am very sorry, **I was doing some stupid and made wrong claims**. The `generation_kwargs` is an argument to the `__init__` of `GenerationConfig` instead of the `generate` method. 

(Some of) The previous failing tests pass as the 2 problematic arguments to `generate` are not passed, but this is wrong logic.

I will look in more depth.

**Comment by ydshieh** [comment]
Well, I finally decided  to overwrite `generate` for `TFWhisperForConditionalGeneartion`.

File: src/transformers/models/whisper/modeling_tf_whisper.py
Comment (ydshieh) [inline src/transformers/models/whisper/modeling_tf_whisper.py:1526]: 😭 😭 😭 
😭 😭 😭 
😭 😭 😭

File: tests/models/whisper/test_modeling_tf_whisper.py
Comment (ArthurZucker) [inline tests/models/whisper/test_modeling_tf_whisper.py:782]: Thanks for finally fixing it!

File: tests/models/whisper/test_modeling_tf_whisper.py
Comment (ArthurZucker) [inline tests/models/whisper/test_modeling_tf_whisper.py:1]: Since you are adding support for prompt ids, either add the `test_generate_with_prompt_ids_and_task_and_language` tests, or split the changes to a separate PR, but let's test this before merging! (2 tests are implemented in torch)

**Review by ArthurZucker — COMMENTED** [review]
Thanks! Needs testing for `prompt_ids` otherwise lgtm! Also we don't need part of the legacy things for timestamps since it was never supported in TF (just in case)

File: src/transformers/models/whisper/modeling_tf_whisper.py
Comment (amyeroberts) [inline src/transformers/models/whisper/modeling_tf_whisper.py:1339]:
```suggestion
        seed: Optional[List[int]] = None,
        return_timestamps: Optional[bool] = None,
        task: Optional[str] = None,
        language: Optional[str] = None,
        is_multilingual: Optional[bool] = None,
```

File: src/transformers/models/whisper/modeling_tf_whisper.py
Comment (amyeroberts) [inline src/transformers/models/whisper/modeling_tf_whisper.py:1360]: nit: we typically say "if unset" rather than if `None` in the docstrings


```suggestion
                The sequence used as a prompt for the generation or as model inputs to the encoder. If unset the
```

**Review by amyeroberts — APPROVED** [review]
Thanks for adding the generate logic and fixing this!  

+1 on @ArthurZucker's comment about adding the tests

File: src/transformers/models/whisper/modeling_tf_whisper.py
Comment (ydshieh) [reply src/transformers/models/whisper/modeling_tf_whisper.py:1339]: Thanks. I know - I copied directly from our torch code, me too lazy. Will keep torch code the untouched in this PR though.

File: src/transformers/models/whisper/modeling_tf_whisper.py
Comment (ydshieh) [reply src/transformers/models/whisper/modeling_tf_whisper.py:1526]: Hmm, there is 9 cry face but I only got 4 yes in the reply ..?

File: tests/models/whisper/test_modeling_tf_whisper.py
Comment (ydshieh) [reply tests/models/whisper/test_modeling_tf_whisper.py:1]: done Sir!

**Pull request merged**
